shared_model/builders/protobuf/transport_builder.hpp

```diff
@@ -54,6 +58,38 @@ namespace shared_model {
       SV stateless_validator_;
     };
 
+    template <typename SV>
+    class TransportBuilder<interface::BlockVariantType, SV> {
+     public:
+      TransportBuilder<interface::BlockVariantType, SV>(
+          SV stateless_validator = SV())
+          : stateless_validator_(stateless_validator) {}
+
+      iroha::expected::Result<interface::BlockVariantType, std::string> build(
+          iroha::protocol::Block transport) {
+        if (transport.payload().transactions().size() == 0) {
+          auto result = shared_model::proto::EmptyBlock(transport);
```

Comment:
You make result, but then construct block again in return statement. Maybe create shared pointer here, to avoid extra construction?